### PR TITLE
new analysis_toolkit.py  

### PR DESCRIPTION
**Pull request by anupama-reghunath**
First version of a new toolkit script for analysis.
preselection_cut(show_table=True) returns a flag and produces a table for the candidate in the event.
Sample Output :

```
+----------------------------------+---------------------------+---------------------+-----------------------+
| Parameter                        | Value                     | Pre-selection cut   | Pre-selection Check   |
+==================================+===========================+=====================+=======================+
| Number of candidates in event    | 1                         | ==1                 | True                  |
+----------------------------------+---------------------------+---------------------+-----------------------+
| Time @ decay vertex (ns)         | 827.838314901504          |                     |                       |
+----------------------------------+---------------------------+---------------------+-----------------------+
| Impact Parameter (cm)            | 1.870657075982154         | IP < 250.0 cm       | True                  |
+----------------------------------+---------------------------+---------------------+-----------------------+
| DOCA (cm)                        | 0.1314989058928542        | DOCA < 1 cm         | True                  |
+----------------------------------+---------------------------+---------------------+-----------------------+
| Is within Fiducial Volume?       | True                      | True                | True                  |
+----------------------------------+---------------------------+---------------------+-----------------------+
| Dist2InnerWall (cm)              | 107.77691678708439        | > 5 cm              | True                  |
+----------------------------------+---------------------------+---------------------+-----------------------+
| Dist2VesselEntrance (cm)         | 4427.433487678709         | > 100 cm            | True                  |
+----------------------------------+---------------------------+---------------------+-----------------------+
| Invariant Mass (GeV)             | 0.9924035137310876        |                     |                       |
+----------------------------------+---------------------------+---------------------+-----------------------+
| Daughter Momentum [d1, d2] (GeV) | [10.6655147  17.80364632] | > 1 GeV             | True                  |
+----------------------------------+---------------------------+---------------------+-----------------------+
| Degrees of Freedom [d1, d2]      | [27 27]                   | > 25                | True                  |
+----------------------------------+---------------------------+---------------------+-----------------------+
| Reduced Chi^2 [d1, d2]           | [0.91172176 0.80305711]   | < 5                 | True                  |
+----------------------------------+---------------------------+---------------------+-----------------------+
| Pre-selection passed:            |                           |                     | True                  |
+----------------------------------+---------------------------+---------------------+-----------------------+

```
I've added an `analysis_example.py` in `macro/`, which was used for this output.

### CODE REVIEW

**Comment by olantwin** [comment]
Thanks a lot for this! While I'm sure the details will evolve based on my (and hopefully others') feedback, I really appreciate the initiative!

Would you have time to briefly introduce the current implementation at the meeting Monday?

**Comment by anupama-reghunath** [comment]
Yes, I do hope we can add on more as we go. I can briefly show the implementation on Monday's meeting :)

**Comment by olantwin** [comment]
Maybe we could move the example to a new directly called `examples`.

@egamberini could we run the example in the CI?

File: macro/analysis_example.py [hunk 1]
```diff
@@ -12,95 +12,101 @@
...
-
-        IP_cut = 250
-        preselection_flag = selection.preselection_cut(
-            signal, IP_cut=IP_cut, show_table=True
-        )
-
-        if preselection_flag:
-            print(
-                f"Event:{event_nr} Candidate_index: {candidate_id_in_event} <--passes the pre-selection\n\n"
+def Main_function():
```
Comment (olantwin) [inline macro/analysis_example.py:25]:
```suggestion
def main():
```

Unless you give it a more descriptive name, the convention would be to call it `main`.

File: macro/analysis_example.py [hunk 2]
```diff
@@ -12,95 +12,101 @@
...
+                    f"Event:{event_nr} Candidate_index: {candidate_id_in_event} <--passes the pre-selection\n\n"
+                )
+            else:
+                print(f"Event:{event_nr} Candidate_index: {candidate_id_in_event} \n\n")
+
+    ut.writeHists(hist_dict, "preselectionparameters.root")
 
 
-ut.writeHists(hist_dict, "preselectionparameters.root")
+Main_function()
```
Comment (olantwin) [inline macro/analysis_example.py:112]:
```suggestion
if __name__ == "__main__":
    main()
```

**Review by olantwin — CHANGES_REQUESTED** [review]

**Comment by olantwin** [comment]
One more general comment: 
For new python code, we should probably make a `ship` module, potentially with e.g. an `analysis_tools` submodule. For things that don't yet have a stable interface, we might initially want to hide them in an `experimental` namespace.

**Review by olantwin — APPROVED** [review]
I expect this to still undergo major changes, but merging this first version now, so that people can start using it and give feedback.

**Comment by olantwin** [comment]
Oh, could you please add a line to the change log?

**Review by olantwin — APPROVED** [review]

**Pull request merged**